svn diff -r13185:13186 svn://svn.code.sf.net/p/bochs/code/trunk/bochs > files/patch-x-01

Index: cpu/proc_ctrl.cc
===================================================================
--- cpu/proc_ctrl.cc	(revision 13185)
+++ cpu/proc_ctrl.cc	(revision 13186)
@@ -318,12 +318,17 @@
   for (unsigned n=0; n<CACHE_LINE_SIZE; n += 32) {
     write_virtual_ymmword(i->seg(), eaddr+n, &ymmzero);
   }
-#else
+#elif NX_CPU_LEVEL >= 6
   BxPackedXmmRegister xmmzero;
   xmmzero.clear();
   for (unsigned n=0; n<CACHE_LINE_SIZE; n += 16) {
     write_virtual_xmmword(i->seg(), eaddr+n, &xmmzero);
   }
+#else
+  Bit64u val_64 = 0;
+  for (unsigned n=0; n<CACHE_LINE_SIZE; n += 8) {
+    write_virtual_qword(i->seg(), eaddr+n, val_64);
+  }
 #endif
 }
 
